zipline/algorithm.py

```diff
@@ -114,6 +115,38 @@
 DEFAULT_CAPITAL_BASE = float("1.0e5")
 
 
+log = logbook.Logger("ZiplineLog")
+
+
+def check_asset_lifetime(f):
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        trading_algorithm = args[0]
+        asset = args[1]
+
+        if not isinstance(asset, Asset):
+            raise UnsupportedOrderParameters(
+                msg="Passing non-Asset argument to 'order()' is not supported."
+                    " Use 'sid()' or 'symbol()' methods to look up an Asset."
+            )
+
+        current_dt = trading_algorithm.get_datetime()
+        if asset.auto_close_date:
+            if asset.end_date < current_dt < asset.auto_close_date:
+                # we are between the asset's end date and auto close date,
+                # so warn the user that they can't place an order for this
+                # asset, and return None.
+                log.warn("Cannot place order for {0}, as it has de-listed. "
+                         "Any existing positions for this asset will be "
+                         "liquidated on "
+                         "{1}.".format(asset.symbol, asset.auto_close_date))
+                return None
```

Comment:
Yeah, we ran into this problem with some legacy algorithms.  Since we already have this problem, and this bandaid is likely going to be rarely applied, I'd rather contain scope for now and spend time coming up with a good design.